### PR TITLE
feat: app bundle, DMG and App Store (macOS & iOS) distribution docs

### PR DESCRIPTION
**Pull request by lucasfernog**
Needs #2579

### CODE REVIEW

File: src/content/docs/distribute/app-store.mdx [hunk 1]
```diff
@@ -2,11 +2,313 @@
...
-    text: Stub
-    variant: caution
 ---
 
-import Stub from '@components/Stub.astro';
+import CommandTabs from '@components/CommandTabs.astro';
 
-<Stub />
+App Store is the app marketplace maintained by Apple.
+You can distribute your Tauri app targetting macOS and iOS.
```
Comment (tweidinger) [inline src/content/docs/distribute/app-store.mdx]:
```suggestion
The [Apple App Store] is the app marketplace maintained by Apple.
You can distribute your Tauri app targeting macOS and iOS via this App Store.
```

File: src/content/docs/distribute/app-store.mdx [hunk 2]
```diff
@@ -2,11 +2,313 @@
...
+/>
+
+The `--ios-color` argument defines the background color for the iOS icons.
+
+## Setting up
+
+After enrolling to the Apple Developer program, the first step to distribute your Tauri app in the App Store
+is to register your app in the [App Store Connect][app-store-connect-apps].
+
+Note that the value provided in the _Bundle ID_ field must match the identifier defined in [`tauri.conf.json > identifier`].
```
Comment (tweidinger) [inline src/content/docs/distribute/app-store.mdx]:
```suggestion
:::note
The value provided in the _Bundle ID_ field **must** match the identifier defined in [`tauri.conf.json > identifier`].
:::
```

File: src/content/docs/distribute/app-store.mdx [hunk 3]
```diff
@@ -2,11 +2,313 @@
...
+
+## Build and upload
+
+The Tauri CLI can package your app for macOS and iOS. Running on a macOS machine is a requirement.
+
+:::note
```
Comment (tweidinger) [inline src/content/docs/distribute/app-store.mdx]:
```suggestion
:::caution
```

File: src/content/docs/distribute/app-store.mdx [hunk 4]
```diff
@@ -2,11 +2,313 @@
...
+
+The iOS and macOS apps are uploaded using `altool`, which uses an App Store Connect API key to authenticate.
+
+To create a new API key, open the [App Store Connect's Users and Access page], select the Keys tab, click on the Add button and select a name and the Developer access.
+The APPLE\*API_ISSUER (Issuer ID) is presented above the keys table, and the APPLE_API_KEY_ID is the value on the Key ID column on that table.
+You also need to download the private key, which can only be done once and is only visible after a page reload (the button is shown on the table row for the newly created key).
+The private key file path must be saved as `AuthKey\_<APPLE_API_KEY_ID>.p8` in one of these directories:`<current-working-directory>/private_keys`, `~/private_keys`, `~/.private_keys`or`~/.appstoreconnect/private_keys`.
+
+[App Bundle distribution guide]: /distribute/macos-application-bundle
+[Apple Developer]: https://developer.apple.com
```
Comment (tweidinger) [inline src/content/docs/distribute/app-store.mdx:307]:
```suggestion
[Apple Developer]: https://developer.apple.com
[Apple App Store]: https://www.apple.com/store
[`altool`]: https://help.apple.com/itc/apploader/#/apdATD1E53-D1E1A1303-D1E53A1126
```

File: src/content/docs/distribute/app-store.mdx [hunk 5]
```diff
@@ -2,11 +2,313 @@
...
+
+Note that you must replace _$APPNAME_ with your app name.
+
+:::note
+You must sign the PKG with a _Mac Installer Distribution_ signing certificate.
+:::
+
+#### Upload
+
+Now you can use the `altool` CLI to upload your app PKG to the App Store:
```
Comment (tweidinger) [inline src/content/docs/distribute/app-store.mdx]:
```suggestion
Now you can use the [`altool`] CLI to upload your app PKG to the App Store:
```

File: src/content/docs/distribute/app-store.mdx [hunk 6]
```diff
@@ -2,11 +2,313 @@
...
+See the [authentication section] for more information.
+
+Your app will then be validated by Apple and available in TestFlight if approved.
+
+### Authentication
+
+The iOS and macOS apps are uploaded using `altool`, which uses an App Store Connect API key to authenticate.
+
+To create a new API key, open the [App Store Connect's Users and Access page], select the Keys tab, click on the Add button and select a name and the Developer access.
+The APPLE\*API_ISSUER (Issuer ID) is presented above the keys table, and the APPLE_API_KEY_ID is the value on the Key ID column on that table.
```
Comment (tweidinger) [inline src/content/docs/distribute/app-store.mdx]:
```suggestion
The `APPLE_API_ISSUER` (Issuer ID) is presented above the keys table, and the `APPLE_API_KEY_ID` is the value on the Key ID column on that table.
```

File: src/content/docs/distribute/macos-application-bundle.mdx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+  pnpm="pnpm tauri build --bundles app"
+  cargo="cargo tauri build --bundles app"
+/>
+
+## File structure
+
+The macOS app bundle is a directory with the following structure:
+
+```
+
```
Comment (tweidinger) [inline src/content/docs/distribute/macos-application-bundle.mdx]: newline needed?

File: src/content/docs/distribute/macos-application-bundle.mdx [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+│   ├── Frameworks
+│   ├── PlugIns
+│   ├── SharedSupport
+```
+
+See the [official documentation](https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFBundles/BundleTypes/BundleTypes.html) for more information.
+
+## Native configuration
+
+The app bundle is configured by the Info.plist file, which includes key-value pairs with your app identity and configuration values read by macOS.
```
Comment (tweidinger) [inline src/content/docs/distribute/macos-application-bundle.mdx]:
```suggestion
The app bundle is configured by the `Info.plist` file, which includes key-value pairs with your app identity and configuration values read by macOS.
```

File: src/content/docs/distribute/macos-application-bundle.mdx [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+	<key>NSCameraUsageDescription</key>
+	<string>Request camera access for WebRTC</string>
+	<key>NSMicrophoneUsageDescription</key>
+	<string>Request microphone access for WebRTC</string>
+</dict>
+</plist>
+
+```
+
+This Info.plist file is merged with the values generated by the Tauri CLI. Be careful when overwriting default values such as application version as they might conflict with other configuration values
```
Comment (tweidinger) [inline src/content/docs/distribute/macos-application-bundle.mdx]:
```suggestion
This `Info.plist` file is merged with the values generated by the Tauri CLI. Be careful when overwriting default values such as application version as they might conflict with other configuration values
```

File: src/content/docs/distribute/macos-application-bundle.mdx (hunk 7 above)
Comment (lucasfernog) [reply src/content/docs/distribute/macos-application-bundle.mdx]: oopsy, fixed

**Pull request merged**